backend/core/src/main/java/org/sonarsource/sonarlint/core/websocket/WebSocketService.java

```diff
@@ -95,6 +78,16 @@ public void handleEvent(BindingConfigChangedEvent bindingConfigChangedEvent) {
       return;
     }
     executorService.execute(() -> considerScope(bindingConfigChangedEvent.configScopeId()));
+    // possible change of region for the binding; need to unsubscribe from the old region (subscription to the new one will be done in considerScope)
+    if (didChangeRegion(bindingConfigChangedEvent.previousConfig(), bindingConfigChangedEvent.newConfig())) {
+      executorService.execute(() -> {
```

Comment:
Could making the check before being in the executor cause some problems, like race conditions? You could have everything executed in the call to execute above